dev/is-changed.py

```diff
@@ -65,6 +65,20 @@ def main():
         changed_files = identify_changed_files_from_git_commits(
             os.environ["GITHUB_SHA"], target_ref=os.environ["GITHUB_PREV_SHA"]
         )
+
+    if any(f.endswith(".jar") for f in changed_files):
+        with open(
+            os.path.join(os.path.dirname(os.path.realpath(__file__)), "test-jars.txt")
+        ) as jarlist:
+            inter = set((line.strip() for line in jarlist.readlines())).intersection(
+                set(changed_files)
+            )
+            if len(inter) > 0:
+                raise SystemExit(
+                    "Cannot include jars in source codes. If this has to be added temporarily, "
+                    "please add the file name into dev/test-jars.txt."
```

Comment:
Maybe we should mention that the author needs to add an assumption of the artifact's existence for related tests

Reply:
sg